### PR TITLE
HIVE-27659: Make partition order configurable if the metastore does not return all partitions

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

This PR makes partition order configurable if the metastore does not return all partitions.

### Why are the changes needed?

We want to improve performance of limit-only queries on many partitioned tables.
For example: `SELECT * FROM large_partition_table limit 10`.
But we want to get the latest partition data. After this PR, we can `set metastore.partition.order.expr="PARTITIONS"."CREATE_TIME"` to achieve it.

### Does this PR introduce _any_ user-facing change?

No.

### Is the change a dependency upgrade?

No.

### How was this patch tested?

Unit test.

### CODE REVIEW

**Comment by wangyum** [comment]
cc @sunchao

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java [hunk 1]
```diff
@@ -3959,7 +3959,7 @@ public List<Partition> listPartitionsPsWithAuth(String catName, String db_name,
         LOG.info(
             "Redirecting to directSQL enabled API: db: {} tbl: {} partVals: {}",
             db_name, tbl_name, Joiner.on(',').join(part_vals));
-        return getPartitions(catName, db_name, tbl_name, -1);
+        return getPartitions(catName, db_name, tbl_name, max_parts);
```
Comment (sunchao) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java:3962]: Is this change related?

Comment (wangyum) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java:3962]: Will fix it in a separate PR: https://github.com/apache/hive/pull/4662

Comment (wangyum) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java]: Fixed.

**Review by sunchao — APPROVED** [review]
LGTM

**Comment by sunchao** [comment]
Merged, thanks @wangyum !
